packages/backend/src/dbt/dbtChildProcess.ts

```diff
@@ -0,0 +1,65 @@
+import execa from "execa";
+import {ChildProcess} from "child_process";
+
+export class DbtChildProcess {
+    static host: string = 'localhost';
+    port: number;
+    dbtChildProcess: undefined | ChildProcess;
+    processRunning: boolean;
+    projectDir: string;
+    profilesDir: string;
+
+    constructor(projectDir: string, profilesDir: string, port: number) {
+        this.port = port
+        this.processRunning = false;
+        this.projectDir = projectDir;
+        this.profilesDir = profilesDir;
+    }
+
+    private _start () {
+        this.dbtChildProcess = execa(
+            'dbt',
+            ['rpc', '--host', DbtChildProcess.host, '--port', `${this.port}`, '--profiles-dir', this.profilesDir],
+            {
+                cwd: this.projectDir,
+                stdio: ['pipe', 'pipe', process.stderr]
+            }
+        );
+
+        this.dbtChildProcess.on('spawn', () => {
+            console.log('Lightdash started a dbt subprocess');
+            this.processRunning = true;
+        });
+
+        this.dbtChildProcess.on('exit', (exitCode, killSignal) => {
+            this.processRunning = false;
+            if (exitCode !== 2) {
+                console.log('The dbt subprocess exited unexpectedly. Lightdash is restarting it');
+                this._start();
+            }
+            console.log('The dbt subprocess exited');
+        });
+
+        if (this.dbtChildProcess.stdout) {
+            this.dbtChildProcess.stdout.on('data', (data) => {
+                try {
+                    const message = data.toString('utf8');
+                    const payload = JSON.parse(message);
+                    console.log(payload);
+                }
+                catch (e) {
+                    console.log('Cannot parse message from dbt');
+                }
+            })
+        }
+    }
+
+    public refresh () {
+        if (this.processRunning) {
+            this.dbtChildProcess && this.dbtChildProcess.kill(1) // send SIGHUP
+        }
+        else
+            this._start()
+    }
```

Comment:
On refresh, are we suppose to kill the existing process and start again ? right now is a if/else so on refresh we just kill the process.

Reply:
there's an event on the subprocess that restarts the process if it exits from a kill signal

Reply:
ok, looks like we have the restart logic inside the `exit` handler. Not sure how I feel about that.
Wondering if we should turn these into promises. 

```
    private async _start () {
        return new Promise<void>((resolve, reject) => {
        ...
            this.dbtChildProcess.on('spawn', () => {
                ...
                resolve();
            });

            this.dbtChildProcess.on('exit', (exitCode, killSignal) => {
                ...
                reject();
            });
      ...
        });
    }

    public async refresh () {
        if (this.processRunning) {
            this.dbtChildProcess && this.dbtChildProcess.kill(1) // send SIGHUP
        }

        return await this._start();
    }
    ```
    
    Then  in `DbtLocalProjectAdapter` 
    ```
            await this.dbtChildProcess.refresh()
            ```

Reply:
Also, maybe we should turn this file/class into a singleton since we only want one childprocess at any given time.

Reply:
Doesn't that code suggest that the process can only spawn or exit? Usually they both happen